astropy/_erfa/erfa_generator.py

```diff
@@ -388,7 +388,7 @@ def __init__(self, name, source_path, match_line=None):
             else:
                 filecontents = f.read()
 
-        pattern = fr"\n([^\n]+{name} ?\([^)]+\)).+?(/\*.+?\*/)"
+        pattern = fr"\n([^\n=]+{name} ?\([^)]+\)).+?(/\*.+?\*/)"
```

Comment:
This is because `dat.c` now has an `stat = eraDat`, which should not be recognized as the function definition.

Reply:
Gotta say: these regexes are pretty hairy.  To get the PR adding the new ERFA in I spent probably 45 min working out the one-or-two-character changes... (but this one looks good!)

Reply:
Yes, hairy it is...